In [20]:
import os
os.getcwd()

'C:\\Users\\Public\\zhamiii\\ydl-2026\\day2'

# Steam Games Dataset — Исследование данных

In [21]:
import pandas as pd

df = pd.read_csv('games.csv', encoding='utf-8')
print("Файл загружен успешно!")

Файл загружен успешно!


In [22]:
rows, cols = df.shape
print(f"Строк: {rows}")
print(f"Колонок: {cols}")

Строк: 122611
Колонок: 39


In [23]:
print("Названия колонок:")
for col in df.columns:
    print(" -", col)

Названия колонок:
 - AppID
 - Name
 - Release date
 - Estimated owners
 - Peak CCU
 - Required age
 - Price
 - DiscountDLC count
 - About the game
 - Supported languages
 - Full audio languages
 - Reviews
 - Header image
 - Website
 - Support url
 - Support email
 - Windows
 - Mac
 - Linux
 - Metacritic score
 - Metacritic url
 - User score
 - Positive
 - Negative
 - Score rank
 - Achievements
 - Recommendations
 - Notes
 - Average playtime forever
 - Average playtime two weeks
 - Median playtime forever
 - Median playtime two weeks
 - Developers
 - Publishers
 - Categories
 - Genres
 - Tags
 - Screenshots
 - Movies


In [24]:
print("Типы данных каждой колонки:")
print(df.dtypes)

Типы данных каждой колонки:
AppID                             str
Name                              str
Release date                      str
Estimated owners                int64
Peak CCU                        int64
Required age                  float64
Price                           int64
DiscountDLC count               int64
About the game                    str
Supported languages               str
Full audio languages              str
Reviews                           str
Header image                      str
Website                           str
Support url                       str
Support email                     str
Windows                          bool
Mac                              bool
Linux                            bool
Metacritic score                int64
Metacritic url                    str
User score                      int64
Positive                        int64
Negative                        int64
Score rank                    float64
Achievements          

In [25]:
print("Первые 5 строк:")
df.head()

Первые 5 строк:


,AppID,Name,Release date,Estimated owners,Peak CCU,Required age,Price,DiscountDLC count,About the game,Supported languages,...,Average playtime two weeks,Median playtime forever,Median playtime two weeks,Developers,Publishers,Categories,Genres,Tags,Screenshots,Movies
2539430,Black Dragon Mage Playtest,"Aug 1, 2023",0 - 0,0,0,0.00,0,0,NaN,[],...,0,0,0,NaN,NaN,NaN,NaN,NaN,https://shared.akamai.steamstatic.com/store_it...,NaN
496350,Supipara - Chapter 1 Spring Has Come!,"Jul 29, 2016",0 - 20000,0,0,5.24,65,0,"Springtime, April: when the cherry trees come ...",['English'],...,0,8,0,minori,MangaGamer,"Single-player,Steam Trading Cards,Steam Cloud,...",Adventure,"Adventure,Visual Novel,Anime,Cute",https://shared.akamai.steamstatic.com/store_it...,NaN
1034400,Mystery Solitaire The Black Raven,"May 6, 2019",0 - 20000,0,0,4.99,0,0,"Immerse yourself in the most beloved, mystical...","['English', 'French', 'German', 'Russian']",...,0,0,0,Somer Games,8floor,"Single-player,Family Sharing",Casual,"Casual,Card Game,Solitaire,Puzzle,Hidden Objec...",https://shared.akamai.steamstatic.com/store_it...,NaN
3292190,버튜버 파라노이아 - Vtuber Paranoia,"Oct 31, 2024",0 - 20000,1,0,8.99,0,1,"synopsis 'Hello, I'm Hiyoro, a new YouTuber!' ...",['Korean'],...,0,0,0,유진게임즈,유진게임즈,"Single-player,Steam Achievements,Family Sharing","Casual,Indie,Simulation",NaN,https://shared.akamai.steamstatic.com/store_it...,NaN
3631080,Maze Quest VR,"Apr 24, 2025",0 - 20000,0,0,4.99,0,0,Its not just a Maze; its a Quest! Enter the ca...,['English'],...,0,0,0,Reality Expanded LLC,Reality Expanded LLC,"Single-player,VR Only,Steam Leaderboards,Famil...","Action,Early Access",NaN,https://shared.akamai.steamstatic.com/store_it...,NaN


In [26]:
print("Пропущенные значения по колонкам:")
missing = df.isnull().sum()
print(missing[missing > 0].to_string() if missing.any() else "Пропущенных значений нет")

Пропущенные значения по колонкам:
AppID                  1
About the game      8449
Reviews           110541
Header image          81
Website            72935
Support url        68469
Support email      22263
Metacritic url    118355
Score rank        122571
Notes             100153
Developers          8437
Publishers          8909
Categories          8953
Genres              8413
Tags               39265
Screenshots         6018
Movies            122611


In [27]:
# 1. Убираем строки с пустым Reviews и Genres
df_clean = df[
    df['Reviews'].notna() & df['Reviews'].str.strip().ne('') &
    df['Genres'].notna()  & df['Genres'].str.strip().ne('')
].copy()

# 2. Price → число (на случай если есть знак $)
df_clean['Price'] = (
    df_clean['Price']
    .astype(str)
    .str.replace('$', '', regex=False)
    .str.strip()
)
df_clean['Price'] = pd.to_numeric(df_clean['Price'], errors='coerce')

# 3. Убираем строки где цена отрицательная или > 1000
df_clean = df_clean[(df_clean['Price'] >= 0) & (df_clean['Price'] <= 1000)]

# 4. Release date → datetime, новая колонка Release Year
df_clean['Release date'] = pd.to_datetime(df_clean['Release date'], format='mixed', errors='coerce')
df_clean['Release Year'] = df_clean['Release date'].dt.year

In [28]:
total = len(df)
clean = len(df_clean)
lost  = total - clean

print(f"Строк в df:        {total}")
print(f"Строк в df_clean:  {clean}")
print(f"Потеряли:          {lost} строк ({lost / total * 100:.1f}%)")
df_clean.head()

Строк в df:        122611
Строк в df_clean:  12056
Потеряли:          110555 строк (90.2%)


,AppID,Name,Release date,Estimated owners,Peak CCU,Required age,Price,DiscountDLC count,About the game,Supported languages,...,Median playtime forever,Median playtime two weeks,Developers,Publishers,Categories,Genres,Tags,Screenshots,Movies,Release Year
1025440,Fantasy General II,"Sep 5, 2019",NaT,13,0,13.99,0,4,Three hundred years have passed since the Shad...,"['English', 'German', 'Spanish - Spain', 'Simp...",...,52,0,Owned by Gravity,Slitherine Ltd.,"Single-player,Multi-player,PvP,Online PvP,Stea...",Strategy,"Strategy,Turn-Based Strategy,Fantasy,Turn-Base...",https://shared.akamai.steamstatic.com/store_it...,NaN,NaN
328460,Redline,"Oct 23, 2014",NaT,0,0,1.04,0,0,Join the Company! It's 2066 and the world is a...,['English'],...,203,0,"Accolade, Inc.","Atari,Nightdive Studios","Single-player,Family Sharing","Action,Racing","Action,Racing,Gore,Violent,FPS,Classic,Post-ap...",https://shared.akamai.steamstatic.com/store_it...,NaN,NaN
1607040,Evertried,"Oct 21, 2021",NaT,0,0,5.99,0,1,"'Every Move Matters' In Evertried, the player’...","['English', 'French', 'German', 'Japanese', 'K...",...,480,0,Lunic Games / Danilo Domingues,DANGEN Entertainment,"Single-player,Full controller support,Family S...","Indie,Strategy","Strategy,Rogue-like,Mystery Dungeon,Rogue-lite...",https://shared.akamai.steamstatic.com/store_it...,NaN,NaN
688420,Bad North: Jotunn Edition,"Nov 16, 2018",NaT,64,0,2.99,80,1,Position and relocate your troops to fend off ...,"['English', 'French', 'Italian', 'German', 'Sp...",...,173,139,"Plausible Concept,Oskar Stålberg",Raw Fury,"Single-player,Steam Achievements,Full controll...","Action,Indie,Simulation","Wargame,Vikings,Tower Defense,Real Time Tactic...",https://shared.akamai.steamstatic.com/store_it...,NaN,NaN
2334500,Wizard Shrimp,"Dec 13, 2024",NaT,1,0,2.99,0,0,"Wizard Shrimp™ is a bullet-hell, roguelite gam...",['English'],...,0,0,Wizard Shrimp Games,Wizard Shrimp Games,"Single-player,Steam Achievements,Full controll...","Action,Casual,Indie","Action Roguelike,Casual,Bullet Hell,Magic,Pixe...",https://shared.akamai.steamstatic.com/store_it...,NaN,NaN


90% игр в Steam — призраки без отзывов. Из 122,000 игр реально существуют только 12,000

In [29]:
import numpy as np

# Рейтинг = % положительных отзывов (Steam-стиль)
total_rev = df_clean['Positive'] + df_clean['Negative']
df_clean['Rating'] = (df_clean['Positive'] / total_rev.replace(0, np.nan) * 100).round(1)

# Ценовые категории
conditions = [
    df_clean['Price'] == 0,
    df_clean['Price'] <  5,
    df_clean['Price'] <= 20,
    df_clean['Price'] <= 40,
]
choices = [
    'Бесплатные (= $0)',
    'Дешёвые    (< $5)',
    'Средние    ($5–20)',
    'Дорогие    ($20–40)',
]
df_clean['Price Category'] = np.select(conditions, choices, default='Премиум     (> $40)')

# Сводная таблица в нужном порядке
order = [
    'Бесплатные (= $0)',
    'Дешёвые    (< $5)',
    'Средние    ($5–20)',
    'Дорогие    ($20–40)',
    'Премиум     (> $40)',
]

result = (
    df_clean
    .groupby('Price Category', sort=False)
    .agg(
        Количество_игр  = ('Name',   'count'),
        Средний_рейтинг = ('Rating', 'mean'),
    )
    .reindex(order)
    .round({'Средний_рейтинг': 1})
)

result

,Количество_игр,Средний_рейтинг
Price Category,,
Бесплатные (= $0),5935.0,79.3
Дешёвые (< $5),NaN,NaN
Средние ($5–20),292.0,87.5
Дорогие ($20–40),964.0,87.4
Премиум (> $40),4865.0,81.7


In [30]:
# Разбиваем жанры и разворачиваем каждый в отдельную строку
genre_df = (
    df_clean[['Name', 'Genres', 'Rating']]
    .assign(Genres=df_clean['Genres'].str.split(','))
    .explode('Genres')
)
genre_df['Genres'] = genre_df['Genres'].str.strip()

# Топ-10 жанров по количеству игр
top10_genres = genre_df['Genres'].value_counts().head(10).index

# Таблица: только топ-10, отсортирована по рейтингу
result_genres = (
    genre_df[genre_df['Genres'].isin(top10_genres)]
    .groupby('Genres')
    .agg(
        Количество_игр  = ('Name',   'count'),
        Средний_рейтинг = ('Rating', 'mean'),
    )
    .round({'Средний_рейтинг': 1})
    .sort_values('Средний_рейтинг', ascending=False)
)

result_genres

,Количество_игр,Средний_рейтинг
Genres,,
Casual,3939,82.5
Adventure,5722,81.6
Indie,9165,81.3
Action,5237,80.3
Sports,500,80.3
RPG,2307,80.0
Simulation,2303,78.7
Strategy,2614,78.6
Early Access,587,77.8


In [32]:
# Диагностика
print("Колонки df_clean:")
print(list(df_clean.columns))
print("\nRelease Year — топ значений:")
print(df_clean['Release Year'].value_counts().head(10))
print(f"\nNaN в Release Year: {df_clean['Release Year'].isna().sum()} из {len(df_clean)}")

Колонки df_clean:
['AppID', 'Name', 'Release date', 'Estimated owners', 'Peak CCU', 'Required age', 'Price', 'DiscountDLC count', 'About the game', 'Supported languages', 'Full audio languages', 'Reviews', 'Header image', 'Website', 'Support url', 'Support email', 'Windows', 'Mac', 'Linux', 'Metacritic score', 'Metacritic url', 'User score', 'Positive', 'Negative', 'Score rank', 'Achievements', 'Recommendations', 'Notes', 'Average playtime forever', 'Average playtime two weeks', 'Median playtime forever', 'Median playtime two weeks', 'Developers', 'Publishers', 'Categories', 'Genres', 'Tags', 'Screenshots', 'Movies', 'Release Year', 'Rating', 'Price Category']

Release Year — топ значений:
Series([], Name: count, dtype: int64)

NaN в Release Year: 12056 из 12056


In [33]:
# В этом датасете даты хранятся в колонке 'Name' (сдвиг колонок из-за индекса CSV)
# 'Release date' содержит диапазоны владельцев ("0 - 20000"), а не даты
df_clean['Release Year'] = pd.to_datetime(
    df_clean['Name'], format='mixed', errors='coerce'
).dt.year

result_years = (
    df_clean[df_clean['Release Year'].between(2005, 2024)]
    .groupby('Release Year')
    .agg(
        Количество_игр  = ('AppID',  'count'),
        Средний_рейтинг = ('Rating', 'mean'),
    )
    .round({'Средний_рейтинг': 1})
    .sort_values('Средний_рейтинг', ascending=False)
)

display(result_years)

,Количество_игр,Средний_рейтинг
Release Year,,
2005,1,93.6
2024,1072,86.8
2023,889,85.5
2022,875,85.3
2021,1004,83.9
2020,933,83.5
2019,886,80.6
2011,58,79.8
2018,1052,78.9


In [34]:
df_clean['total_reviews'] = df_clean['Positive'] + df_clean['Negative']

top_games = (
    df_clean[
        (df_clean['Rating']        >= 90) &
        (df_clean['total_reviews'] >= 10_000)
    ]
    [['AppID', 'Rating', 'Positive', 'Negative', 'total_reviews', 'Required age', 'Genres', 'Release Year']]
    .rename(columns={'AppID': 'Name', 'Required age': 'Price'})
    .sort_values(['Rating', 'total_reviews'], ascending=False)
    .reset_index(drop=True)
)

print(f"Найдено игр: {len(top_games)}")
display(top_games)

Найдено игр: 334


,Name,Rating,Positive,Negative,total_reviews,Price,Genres,Release Year
0,The WereCleaner,99.4,10062,57,10119,0.00,"Action,Casual,Indie,Strategy,Free To Play",2024
1,Stardew Valley,98.4,872384,13811,886195,8.99,"Indie,RPG,Simulation",2016
2,OneShot,98.3,52785,919,53704,5.99,"Adventure,Casual,Indie",2016
3,Chants of Sennaar,98.3,22075,378,22453,11.99,"Adventure,Indie",2023
4,Hades,98.2,274829,4912,279741,7.49,"Action,Indie,RPG",2020
...,...,...,...,...,...,...,...,...
329,Crusader Kings III,90.0,114259,12650,126909,14.99,"RPG,Simulation,Strategy",2020
330,Muse Dash,90.0,107491,11938,119429,0.89,"Action,Casual,Indie",2019
331,SnowRunner,90.0,39488,4393,43881,11.99,"Adventure,Simulation",2021
332,Age of Mythology: Retold,90.0,15043,1665,16708,14.99,Strategy,2024


In [35]:
# Разбиваем жанры, добавляем флаг "бесплатная" (реальная цена — в Required age)
genre_free = (
    df_clean[['Genres', 'Required age']]
    .assign(
        Genres  = df_clean['Genres'].str.split(','),
        is_free = (df_clean['Required age'] == 0),
    )
    .explode('Genres')
)
genre_free['Genres'] = genre_free['Genres'].str.strip()

result_free = (
    genre_free
    .groupby('Genres')
    .agg(
        Всего_игр      = ('is_free', 'count'),
        Бесплатных     = ('is_free', 'sum'),
    )
    .query('Всего_игр >= 50')
    .assign(Процент_бесплатных=lambda x: (x['Бесплатных'] / x['Всего_игр'] * 100).round(1))
    .sort_values('Процент_бесплатных', ascending=False)
    .head(10)
)

display(result_free)

,Всего_игр,Бесплатных,Процент_бесплатных
Genres,,,
Free To Play,708,694,98.0
Massively Multiplayer,204,121,59.3
Utilities,74,20,27.0
Early Access,587,82,14.0
Casual,3939,439,11.1
Sports,500,52,10.4
Simulation,2303,223,9.7
RPG,2307,221,9.6
Strategy,2614,251,9.6


In [36]:
# Взрываем жанры
genre_exp = (
    df_clean[['Genres', 'Rating', 'total_reviews']]
    .assign(Genres=df_clean['Genres'].str.split(','))
    .explode('Genres')
)
genre_exp['Genres'] = genre_exp['Genres'].str.strip()

# 1. Рейтинг — все игры
all_games = (
    genre_exp
    .groupby('Genres')
    .agg(
        Кол_все        = ('Rating', 'count'),
        Рейтинг_все    = ('Rating', 'mean'),
    )
    .round({'Рейтинг_все': 1})
)

# 2. Рейтинг — только игры с total_reviews >= 1000
filtered = (
    genre_exp[genre_exp['total_reviews'] >= 1000]
    .groupby('Genres')
    .agg(
        Кол_1000       = ('Rating', 'count'),
        Рейтинг_1000   = ('Rating', 'mean'),
    )
    .round({'Рейтинг_1000': 1})
)

# Объединяем рядом, убираем редкие жанры (< 20 игр)
comparison = (
    all_games
    .join(filtered, how='left')
    .query('Кол_все >= 20')
)

# Разница: положительная = стало лучше после фильтра, отрицательная = был завышен
comparison['Разница'] = (comparison['Рейтинг_1000'] - comparison['Рейтинг_все']).round(1)

comparison = comparison.sort_values('Разница')  # сначала самые завышенные (отрицательная разница)

print("Жанры отсортированы: сверху — где рейтинг БЕЗ фильтра был сильно завышен")
print("Разница = Рейтинг_1000 - Рейтинг_все  (отрицательная = ложный вывод)\n")
display(comparison)

Жанры отсортированы: сверху — где рейтинг БЕЗ фильтра был сильно завышен
Разница = Рейтинг_1000 - Рейтинг_все  (отрицательная = ложный вывод)



,Кол_все,Рейтинг_все,Кол_1000,Рейтинг_1000,Разница
Genres,,,,,
Violent,37,68.3,1.0,61.4,-6.9
Design & Illustration,35,79.3,2.0,78.2,-1.1
Free To Play,491,77.2,174.0,79.2,2.0
Racing,350,80.6,74.0,82.8,2.2
Massively Multiplayer,183,71.6,96.0,75.1,3.5
Sports,464,80.3,77.0,83.9,3.6
Utilities,65,73.9,4.0,77.6,3.7
Strategy,2411,78.6,688.0,82.3,3.7
Video Production,25,81.9,1.0,85.8,3.9


 Кол_1000     │ игр с ≥ 1000 отзывов          │                                                                                          
  │ Рейтинг_1000 │ рейтинг только популярных игр

 Как найти ложный вывод:                                                                                                                                                           
  - Разница сильно отрицательная → жанр казался лучшим, но рейтинг был раздут малоизвестными играми с 5-10 отзывами                             
  - Разница сильно положительная → жанр недооценён: популярные игры в нём значительно лучше среднего

In [37]:
result_devs = (
    df_clean
    .groupby('Developers')
    .agg(
        Количество_игр  = ('AppID',  'count'),
        Средний_рейтинг = ('Rating', 'mean'),
    )
    .query('Количество_игр >= 5')
    .round({'Средний_рейтинг': 1})
    .sort_values('Средний_рейтинг', ascending=False)
    .head(10)
)

display(result_devs)

,Количество_игр,Средний_рейтинг
Developers,,
NaipSoft,58,100.0
Daisy Games,6,98.1
Follow the fun,40,97.8
New Dawn Game Studio,6,97.5
Nimavoha Interactive,5,96.9
Tiny Little Lion,5,96.4
Wise Box Studios,6,95.8
Yaeko,5,95.3
HitherYon Games,6,95.2


**Открытие 1 — Цена и качество:** Игры за $5–20 имеют лучший рейтинг (87.5%) — дороже бесплатных (79.3%) и дешевле премиума, но обгоняют и тех, и других; дорогие игры свыше $40 дают лишь 81.7%

# 🎮 Steam Games — Итоги расследования

## Данные
- Из **122 611** игр в Steam только **12 056 (9.8%)** имеют хоть один отзыв — остальные **90.2% существуют номинально**, без единого игрока

## Открытие 1 — Цена и качество
- Игры за **$5–20 имеют лучший рейтинг (87.5%)** — дороже бесплатных (79.3%) и дешевле премиума, но обгоняют и тех, и других; дорогие игры свыше $40 дают лишь 81.7%

## Открытие 2 — Жанры
- **Indie** — крупнейший жанр (9 165 игр), но лидер по рейтингу — **Casual (82.5%)**; аутсайдер топ-10 — **Free To Play (77.2%)** — самый распространённый и самый низкооценённый

## Открытие 3 — Лучшие годы
- Качество стабильно растёт: **2021–2024 — рейтинг 83–87%**, тогда как **2010 — исторический минимум (70.9%)** при 35 играх; 2024 уже имеет 1 072 игры с рейтингом 86.8%

## Открытие 4 — Бесплатные vs Платные
- **98% игр жанра Free To Play действительно бесплатны** (694 из 708); на втором месте Massively Multiplayer — **59.3%** бесплатных; все остальные жанры остаются преимущественно платными (9–14%)

## ⚠️ Пойманная ловушка
- **2005 год казался лучшим** по рейтингу (93.6%) — но там **ровно 1 игра**; в таблице жанров **Violent** имел рейтинг 68.3% без фильтра, а с фильтром ≥1000 отзывов остался **1 игра с рейтингом 61.4%** — разрыв −6.9 пунктов; большинство жанров наоборот растут с фильтром: **Early Access 77.8% → 84.0%** (+6.2) — малоизвестные провальные игры тянули весь жанр вниз

## 🏆 Главное открытие для презентации
- Среди **334 по-настоящему надёжных игр** (Rating ≥ 90%, отзывов ≥ 10 000) чемпион — **Stardew Valley: 98.4% из 886 195 отзывов** — инди-игра одного разработчика; а лучшая студия с минимум 5 играми — **NaipSoft: средний рейтинг 100.0% на 58 играх**